docs/proposals/odo-v3-cli/odo-dev.md

```diff
@@ -0,0 +1,91 @@
+# `odo dev`
+
+[#5299](https://github.com/redhat-developer/odo/issues/5298)
+
+Running the application on the cluster for **development** (inner loop)
+
+The result of successful execution of `odo dev` is Devfile component deployed in inner loop mode.
+**By default command deletes all resources that it created before it exits.**
+
+When users runs command with `--cleanup=false` there will be no cleanup before existing. The message "Press Ctrl+c to exit and clean up resources from cluster." will be only "Press Ctrl+c to exit."
+
+If user executed `odo dev --cleanup=false` and then run this command again. The first line of the output should display warning: "Reusing already existing resources".
+
+When there is no devfile in the current directory and no flags were provided command starts in interactive mode to guide user through the devfile selection.
+
+When some flags were provided and there is no devfile in the current directory command exits with error:
+```
+No devfile.yaml in the current directory. Use `odo init component` to get devfile.yaml for your application."
+```
+
+When devfile exists in the current directory deploy application in inner loop mode using the information form `devfile.yaml` in the current directory.
+
+
+`odo dev` commands sets up port-forwarding 
+localhost.
+
+
+
+## Flags
+
+- `-o` (string) output information in a specified format (json).
+- `--watch` (boolean) Run command in watch mode. In this mode command is watching the current directory for file changes and automatically sync change to the container where it rebuilds and reload the application.
```

Comment:
If `--watch=false`, then why do we need to continuously run `odo dev`? Wouldn't it just function like `odo push`?

Reply:
yes